source/common/network/io_handle_impl.h

```diff
@@ -0,0 +1,55 @@
+#include "envoy/network/io_handle.h"
+
+#include "common/api/os_sys_calls_impl.h"
+
+namespace Envoy {
+namespace Network {
+
+class FdIoHandleImpl : public virtual IoHandle {
+public:
+  explicit FdIoHandleImpl(int fd);
+
+  ~FdIoHandleImpl() override{};
+
+  Api::SysCallSizeResult readv(const iovec* iovec, int num_iovec) override;
+
+  Api::SysCallSizeResult writev(const iovec* iovec, int num_iovec) override;
+
+  Api::SysCallIntResult close() override;
+
+  bool isClosed() override;
```

Comment:
count be `bool isClosed() const override;`

Reply:
Technically, most methods (all that do not change the `fd_`) could be marked const, but maybe should do so only on methods that do not modify the underlying socket.

Obviously this change should be made on the base class first.